lib/money/money.rb

```diff
@@ -454,7 +482,7 @@ def with_currency(new_currency)
     if !new_currency || currency == new_currency
       self
     else
-      self.class.new(fractional, new_currency, bank)
+      self.class.new(fractional, new_currency, bank, infinite_precision?)
```

Comment:
Is that bad? I believe that, as being the fourth argument, it's kind of more of "internal" usage. And usually no one would want to pass it explicitly, and instead, just convert new Money instance to a needed "type".

It would also be cool to have them subclasses, but I do not know how to make it convenient to use.

Also, as it's going to be on top of v7, how about to make all the arguments after "currency" optional keyword arguments?